src/Cache/CacheEngine.php

```diff
@@ -171,31 +110,6 @@ public function readMany($keys)
      */
     abstract public function delete($key);
 
-    /**
-     * Delete all keys from the cache
-     *
-     * @param bool $check if true will check expiration, otherwise delete all
-     * @return bool True if the cache was successfully cleared, false otherwise
-     */
-    abstract public function clear($check);
```

Comment:
I'm not sure we can change the `clear()` signature without breaking compatibility as it could have been implemented by userland engines.

Reply:
I was thinking some mote about this last night. We could add the new methods required by the PSR interfaces now, but not implement the actual interfaces until 4.0 when we can make the breaking changes to clear().

Reply:
@markstory In that case wouldn't it be better to also not change the return value for engines in case of cache miss from current `false` to `null`? It could potentially cause issues with current applications doing strict checks.

Reply:
I don't think we should change return values for the existing write/read methods as it could cause compatibility issues. However, we sould implement get(), set(), and has() in a way that will be compatible with the PSR interfaces.